### PR TITLE
#8 invoice模块初始化完成

### PR DESCRIPTION
**Pull request by SuperMrZhang**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/suparking/shared-trading-system/pulls).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by SuperMrZhang** [comment]
#8

File: shared-trad-order-serv/shared-trad-order-impl/src/main/resources/logback-spring.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <!--output to file-->
+    <appender name="file" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>log/order.log</file>
+        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+            <!-- rollover daily -->
+            <fileNamePattern>log/user.%d{yyyy-MM-dd}.%i.log</fileNamePattern>
+            <!-- each file should be at most 100MB, keep 60 days worth of history, but at most 20GB -->
+            <maxFileSize>100MB</maxFileSize>
+            <maxHistory>180</maxHistory>
+            <totalSizeCap>20GB</totalSizeCap>
```
Comment (nuo-promise) [inline shared-trad-order-serv/shared-trad-order-impl/src/main/resources/logback-spring.xml:23]: maxFileSize -> 256MB
maxHistory  -> 30
totalSizeCap -> 5GB

File: shared-trad-invoice-serv/shared-trad-invoice-impl/src/main/java/cn/suparking/invoice/Application.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+@SpringBootApplication
+@EnableTransactionManagement
+@ComponentScan(basePackages = {"cn.suparking"})
+@EnableDiscoveryClient
+@EnableFeignClients(basePackages = {"cn.suparking"})
```
Comment (nuo-promise) [inline shared-trad-invoice-serv/shared-trad-invoice-impl/src/main/java/cn/suparking/invoice/Application.java:14]: delete

File: shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/Application.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+@SpringBootApplication
+@EnableTransactionManagement
+@ComponentScan(basePackages = {"cn.suparking"})
+@EnableDiscoveryClient
+@EnableFeignClients(basePackages = {"cn.suparking"})
```
Comment (nuo-promise) [inline shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/Application.java:15]: delete

File: shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        log.warn("constraint violation exception", e);
+        Set<ConstraintViolation<?>> violations = e.getConstraintViolations();
+        return SpkCommonResult.error(violations.stream()
+                .map(v -> v.getPropertyPath().toString().concat(": ").concat(v.getMessage()))
+                .collect(Collectors.joining("| ")));
+    }
+
+    @ExceptionHandler(SpkCommonException.class)
+    protected SpkCommonResult handleShenyuException(final SpkCommonException exception) {
+        log.error("shenyu admin exception ", exception);
```
Comment (nuo-promise) [inline shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java:102]: change shenyu admin -> 对应模块的名称

**Review by nuo-promise — COMMENTED** [review]
modify changes.

File: shared-trad-order-serv/shared-trad-order-impl/src/main/resources/application.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  #  application:
+  # 定义项目名称
+  #    name: coupon-customer-serv
+  # 走 Nacos config
+  datasource:
+    # mysql数据源
+    username: SPKRoot
+    password: SPKRoot2017!
+    url: jdbc:mysql://192.168.1.105:3306/spk_order?autoReconnect=true&useUnicode=true&characterEncoding=utf8&useSSL=false&allowPublicKeyRetrieval=true&zeroDateTimeBehavior=convertToNull&serverTimezone=UTC
+    #      type: com.zaxxer.hikari.HikariDataSource
```
[the remark this answers is not in the document]
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/resources/application.yml]: modify

File: shared-trad-order-serv/shared-trad-order-impl/src/main/resources/application.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      max-lifetime: 60000
+      auto-commit: true
+  #   cloud-nacos 配置
+  cloud:
+    nacos:
+      discovery:
+        # 服务发现默认开关打开
+        enabled: true
+        # 服务地址,集群用逗号分开
+        server-addr: center.suparking.cn:21310
```
[the remark this answers is not in the document]
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/resources/application.yml]: modify

File: shared-trad-order-serv/shared-trad-order-impl/src/main/resources/bootstrap.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+spring:
+  # 必须把name属性从application.yml迁移出来,否则无法动态刷新
+  application:
+    name: shared-trad-order-serv
+  cloud:
+    nacos:
+      config:
+        enabled: true
+        # nacos config服务器的地址
+        server-addr: center.suparking.cn:21310
```
[the remark this answers is not in the document]
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/resources/bootstrap.yml]: modify

File: shared-trad-order-serv/shared-trad-order-impl/src/main/resources/logback-spring.xml (hunk 1 above)
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/resources/logback-spring.xml]: modify

File: shared-trad-invoice-serv/shared-trad-invoice-impl/src/main/java/cn/suparking/invoice/Application.java (hunk 2 above)
Comment (SuperMrZhang) [reply shared-trad-invoice-serv/shared-trad-invoice-impl/src/main/java/cn/suparking/invoice/Application.java]: delete line 14

File: shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/Application.java (hunk 3 above)
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/Application.java]: delete line 14

File: shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java (hunk 4 above)
Comment (SuperMrZhang) [reply shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java]: modify

File: shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java [hunk 8]
```diff
@@ -99,7 +99,7 @@ protected SpkCommonResult handleConstraintViolationException(final ConstraintVio
 
     @ExceptionHandler(SpkCommonException.class)
     protected SpkCommonResult handleShenyuException(final SpkCommonException exception) {
-        log.error("shenyu admin exception ", exception);
+        log.error("shared order admin exception ", exception);
```
Comment (nuo-promise) [inline shared-trad-order-serv/shared-trad-order-impl/src/main/java/cn/suparking/invoice/exception/ExceptionHandlers.java:102]: admin delete

**Review by nuo-promise — COMMENTED** [review]
modify

**Pull request merged**